### ISSUE TITLE
IllegalArgumentException when trying to open app

### ISSUE THREAD

**Issue by gbakeman** [shrimqy/Sekia#3]
Hello,

I heard about this app recently and wanted to give it a try. I've just finished installing it on my PC, and now my Android phone (OnePlus 6T on Android 9). When I try to open it, it immediately crashes with the following error:

```
FATAL EXCEPTION: DefaultDispatcher-worker-4
Process: com.komu.sekia, PID: 5462
java.lang.IllegalArgumentException: listener already in use
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:47)
	at android.net.nsd.NsdManager.putListener(NsdManager.java:448)
	at android.net.nsd.NsdManager.discoverServices(NsdManager.java:580)
	at komu.seki.data.services.NsdService$startDiscovery$1.invokeSuspend(NsdService.kt:91)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@5633491, Dispatchers.IO]
FATAL EXCEPTION: DefaultDispatcher-worker-1
Process: com.komu.sekia, PID: 5662
java.lang.IllegalArgumentException: listener already in use
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:47)
	at android.net.nsd.NsdManager.putListener(NsdManager.java:448)
	at android.net.nsd.NsdManager.discoverServices(NsdManager.java:580)
	at komu.seki.data.services.NsdService$startDiscovery$1.invokeSuspend(NsdService.kt:91)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@40d8d1b, Dispatchers.IO]
```

Thanks in advance!

**Comment by shrimqy** [comment]
Thank you for trying, would you mind checking which version of the app you're using?

**Comment by gbakeman** [comment]
I apologize for not providing that information before. I noticed that you had just released the 0.3.1 hotfix yesterday - I've updated to that release, and the app is launching on my phone now.

Thank you so much for the quick response, and my best wishes to you for your future work on this app.
